keps/sig-storage/3751-volume-attributes-class/README.md

```diff
@@ -437,46 +437,42 @@ Operation metrics from [csiOperationsLatencyMetric](https://github.com/kubernete
     The **vac_finalizer_controller**:
-    * If the VolumeAttributesClassName is nil or empty, the VolumeAttributesClass will not be added to the vac_finalizer_controller cache
-    * Check if the VolumeAttributesClass exists. If not, the VolumeAttributesClass will not be added to the vac_finalizer_controller cache
+    * If the VolumeAttributesClassName is nil or empty, do nothing.
+    * Check if the VolumeAttributesClass exists. If not, do nothing.
     * Check if this VolumeAttributesClass already has a protection finalizer
     * Add the finalizer to the VolumeAttributesClass if there is none 
-2. PVC created with a VolumeAttributesClass being deleted
-    The **vac_finalizer_controller**:
-    * Check VolumeAttributesClass is being deleted and PVC creation failed
-3. PVC updated to a different VolumeAttributesClass
-    * The **vac_finalizer_controller** will remove finalizer in the VolumeAttributesClass only if after listing all the PVCs/PVs and confirm that this PVC/PV is the last one that is consuming the VolumeAttributesClass in the vac_finalizer_controller cache
+2. PVC updated to a different VolumeAttributesClass
+    * The **vac_finalizer_controller** will remove finalizer in the VolumeAttributesClass only if after listing all the PVCs/PVs from informers via a custom indexer and confirm that this PVC/PV is the last one that is consuming the VolumeAttributesClass in the vac_finalizer_controller cache
     * Check if the new VolumeAttributesClass already has a protection finalizer
-    * Add the finalizer to the new VolumeAttributesClass if there is none 
-4. PVC updated to a different VolumeAttributesClass that is being deleted
-    The **vac_finalizer_controller**:
-    * Check VolumeAttributesClass is being deleted and PVC update failed
-5. PVC has a VolumeAttributesClass and this PVC is deleted
-    * The **vac_finalizer_controller** will remove finalizer in the VolumeAttributesClass only if after listing all the PVCs/PVs and confirm that this PVC/PV is the last one that is consuming the VolumeAttributesClass in the informer(a cache of VolumeAttributesClass) **only**
-6. Delete a VolumeAttributesClass while there is **kubernetes.io/vac-protection** finalizer associated with this VolumeAttributesClass
+    * Add the finalizer to the new VolumeAttributesClass if there is nones
+3. PVC has a VolumeAttributesClass and this PVC is deleted
+    * The **vac_finalizer_controller** will remove finalizer in the VolumeAttributesClass only if after listing all the PVCs/PVs from informers via a custom indexer and confirm that this PVC/PV is the last one that is consuming the VolumeAttributesClass in the informer(a cache of VolumeAttributesClass) **only**
+4. Delete a VolumeAttributesClass while there is **kubernetes.io/vac-protection** finalizer associated with this VolumeAttributesClass
     * Deletion will not return an error but it will add a deletionTimestamp and wait for the finalizer being removed, then remove the VolumeAttributesClass
```

Comment:
Note that the controller is only responsible for deleting the finalizer. The API server will delete the VAC once the finalizer is removed.